app/routes/upload.step3.tsx

```diff
@@ -1,5 +1,37 @@
+import { LoaderFunctionArgs } from '@remix-run/node'
+import { useLoaderData } from '@remix-run/react'
+import { formCookie } from '~/lib/cookies'
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const cookieHeader = request.headers.get('Cookie')
+  const cookieData = await formCookie.parse(cookieHeader)
+
+  return { cookieData }
+}
+
 const step3 = () => {
-  return <div>ImproveResume</div>
+  // @ts-ignore
+  const { cookieData } = useLoaderData()
+
+  return (
+    <>
```

Comment:
**suggestion**: Consider adding a loading state or message while the cookie data is being fetched to improve user experience.